stan/math/rev/core/var.hpp

```diff
@@ -138,6 +138,7 @@ class var_value {
    * @return Adjoint for this variable.
    */
   inline auto& adj() { return vi_->adj_; }
+  inline auto& adj_op() { return vi_->adj_; }
```

Comment:
Why doesn't just `adj()` work?

Reply:
It's related to the below, For the Matrix<var> impl we want a `CWiseUnaryOp` where right now `.adj()` returns a `CWiseUnaryView<>` that fails at compile time with the const error from below

https://stackoverflow.com/questions/56635122/using-coeffref-with-const-cwiseunaryview-failing-when-multiplying-two-cwiseuna

Reply:
But the code here is the same?

Reply:
Oh yes this one is just here so we can call the same code for `matrix<var>` and `var<matrix>`. Checkout the code in `eigen_plugins.h`

Reply:
Yeah that was a fun rabbithole I went down when doing the Eigen plugin stuff. So the issue here is that with a non-const ```Matrix<var>```, calling ```.adj()``` returns a ```CWiseUnaryView``` but you want to return a ```CWiseUnaryOp``` regardless of const-ness to avoid that multiplication bug.

I guess the question here is whether it's better to have a separate member function (i.e., ```adj``` & ```adj_op```), or add constness to the object before the member function gets called.

C++17 would make this easy with ```as_const```:
```
std::as_const(MatVar).adj()
```
But could also use something like (completely untested):
```
const_cast<const decltype(MatVar)>(MatVar).adj()
```

Reply:
I kind of like the separate function approach, though we could have both and just backport `as_const`

Reply:
Yeah the separate function is definitely less verbose! No preference on my end

Reply:
Oh I see something different is happening for something other than this. This is just doing the same thing either way cause that's what needs to happen.